src/Main.elm

```diff
@@ -1,10 +1,226 @@
+            let
+                filteredDir =
+                    filter
+                        (filterBySelectedFilter
+                            (parentAndTypeAre (Just folder.id) model.filterBy)
+                            model.filterBy
+                        )
+                        parsedDirectory
+            in
+            { model
+                | directory = filteredDir
+                , parents = folder.parent :: model.parents
+            }
+
+        GoUp ->
+            let
+                filteredDir =
+                    filter
+                        (filterBySelectedFilter
+                            (parentAndTypeAre (join <| head model.parents) model.filterBy)
+                            model.filterBy
+                        )
+                        parsedDirectory
+            in
+            { model
+                | directory = filteredDir
+                , parents = drop 1 model.parents
+            }
+
+        ChangeFilter at ->
+            let
+                filteredDir =
+                    filter (filterBySelectedFilter (parentAndTypeAre Nothing at) at) parsedDirectory
+            in
+            { directory = filteredDir
+            , parents = []
+            , filterBy = at
+            }
+
+
+showAudienceType : AudienceType -> String
+showAudienceType at =
+    case at of
+        Authored ->
+            "Authored"
+
+        Shared ->
+            "Shared"
+
+        Curated ->
+            "Curated"
+
+
+filterButtonView : AudienceType -> AudienceType -> Html Msg
+filterButtonView at filter =
+    button
+        [ classList [ ( "selected", at == filter ) ]
+        , disabled <| at == filter
+        , onClick <| ChangeFilter at
+        ]
+        [ text <| showAudienceType at ]
+
+
+filterPanelView : Model -> Html Msg
+filterPanelView model =
+    table []
+        [ tr []
+            [ th []
+                [ filterButtonView Authored model.filterBy
+                ]
+            , th []
+                [ filterButtonView Shared model.filterBy
+                ]
+            , th []
+                [ filterButtonView Curated model.filterBy
+                ]
+            ]
+        ]
+
+
+audienceView : Audience -> Html Msg
+audienceView a =
+    div
+        [ classList [ ( "dir-elem", True ), ( "file", True ) ]
+        ]
+        [ text <| a.name ++ " - [" ++ showAudienceType a.type_ ++ "]"
+        ]
+
+
+audienceFolderView : AudienceFolder -> Html Msg
+audienceFolderView af =
+    div
+        [ onClick <| OpenFolder af
+        , classList [ ( "dir-elem", True ), ( "folder", True ) ]
+        ]
+        [ text <| "🗀 Folder: " ++ af.name ]
+
+
+dirElemView : DirElem -> Html Msg
+dirElemView de =
+    case de of
+        File a ->
+            audienceView a
+
+        Folder af ->
+            audienceFolderView af
+
+
+goUpView : Model -> List (Html Msg)
+goUpView model =
+    if length model.parents > 0 then
+        [ div
+            [ onClick GoUp
```

Comment:
I prefer usage of specific state in view (like goTo "someId or index") since it prevent user from doing two events (go up twice) by double click as accident. It's again not big deal, but just food for thoughts :)